src/careamics/dataset/in_memory_dataset.py

```diff
@@ -47,8 +47,8 @@ class InMemoryDataset(Dataset):
     def __init__(
         self,
         data_config: DataConfig,
-        inputs: Union[np.ndarray, List[Path]],
-        input_target: Optional[Union[np.ndarray, List[Path]]] = None,
+        inputs: Union[np.ndarray, list[Path]],
+        input_target: Optional[Union[np.ndarray, list[Path]]] = None,
```

Comment:
Maybe should be `numpy.array` not `np.array`.

Also the [`numpy.typing`](https://numpy.org/devdocs/reference/typing.html#) package had the [`NDArray`](https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray) and [`ArrayLike`](https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike) types for type hinting.

Reply:
Yes that would make a lot of sense!

I would tend to use `NDArray` because `ArrayLike` is a large Union that includes int and such.